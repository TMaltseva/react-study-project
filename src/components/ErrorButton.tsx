import React, { useState } from 'react';

const ErrorButton: React.FC = () => {
  const [throwError, setThrowError] = useState(false);

  if (throwError) {
    throw new Error('Test error from ThrowErrorButton');
  }

  return (
    <button className="throw-error-btn" onClick={() => setThrowError(true)}>
      Throw Error
    </button>
  );
};

export default ErrorButton;
