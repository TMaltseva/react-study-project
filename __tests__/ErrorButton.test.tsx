import renderer from 'react-test-renderer';
import ErrorButton from '../src/components/ErrorButton';

test('ErrorButton component renders correctly', () => {
  const tree = renderer.create(<ErrorButton />).toJSON();
  expect(tree).toMatchSnapshot();
});