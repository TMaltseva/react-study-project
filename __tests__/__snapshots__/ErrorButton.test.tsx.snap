// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorButton component renders correctly 1`] = `
<button
  className="throw-error-btn"
  onClick={[Function]}
>
  Throw Error
</button>
`;
